rabbus.go

```diff
@@ -224,9 +226,12 @@ func (r *rabbus) register() {
 }
 
 func (r *rabbus) produce(m Message) {
-	if err := r.ch.ExchangeDeclare(m.Exchange, m.Kind, r.config.Durable, false, false, false, nil); err != nil {
-		r.emitErr <- err
-		return
+	if _, ok := r.exDeclared[m.Exchange]; !ok {
```

Comment:
I think we didn't need the lock now because the  `func (r *rabbus) produce(m Message)` is only used by a single goroutine and this function is the only function using the map. WDYT?